core/resource/service/resource_service.go

```diff
@@ -146,51 +137,68 @@ func (rs ResourceService) Deploy(ctx context.Context, tnnt tenant.Tenant, store
-		rs.logger.Error("error reading all existing resources: %s", err)
-	}
+	toUpdateOnStore, err := rs.getResourcesToBatchUpdate(ctx, tnnt, store, resources)
+	multiError.Append(err)
 
-	existingMappedByFullName := rs.getResourcesMappedByFullName(existingResources)
-	resourcesToBatchUpdate := rs.getResourcesToBatchUpdate(resources, existingMappedByFullName)
-	if len(resourcesToBatchUpdate) == 0 {
+	if len(toUpdateOnStore) == 0 {
 		rs.logger.Warn("no resources to be batch updated")
+		return errors.MultiToError(multiError)
 	}
 
-	multiError.Append(rs.batch.CreateOrUpdateAll(ctx, resourcesToBatchUpdate))
-	multiError.Append(rs.mgr.BatchUpdate(ctx, store, resourcesToBatchUpdate))
+	multiError.Append(rs.mgr.BatchUpdate(ctx, store, toUpdateOnStore))
 	return errors.MultiToError(multiError)
 }
 
-func (rs ResourceService) getResourcesToBatchUpdate(incomings []*resource.Resource, existingMappedByFullName map[string]*resource.Resource) []*resource.Resource {
-	var output []*resource.Resource
+func (rs ResourceService) getResourcesToBatchUpdate(ctx context.Context, tnnt tenant.Tenant, store resource.Store, incomings []*resource.Resource) ([]*resource.Resource, error) {
+	existingResources, readErr := rs.repo.ReadAll(ctx, tnnt, store)
+	if readErr != nil {
+		rs.logger.Error("error reading all existing resources: %s", readErr)
+		return nil, readErr
+	}
+
+	existingMappedByFullName := createFullNameToResourceMap(existingResources)
+
+	var toUpdateOnStore []*resource.Resource
+	me := errors.NewMultiError("error in resources to batch update")
+
 	for _, incoming := range incomings {
 		if incoming.Status() != resource.StatusValidationSuccess {
 			continue
 		}
-		if existing, ok := existingMappedByFullName[incoming.FullName()]; ok {
-			if incoming.Equal(existing) && existing.Status() == resource.StatusSuccess {
-				incoming.MarkSkipped()
-
-				rs.logger.Warn("resource [%s] is skipped because it has no changes", existing.FullName())
-			} else {
-				incoming.MarkToUpdate()
-				output = append(output, incoming)
 
-				rs.logger.Info("resource [%s] will be updated", existing.FullName())
+		existing, ok := existingMappedByFullName[incoming.FullName()]
+		if !ok {
+			_ = incoming.MarkToCreate()
+			err := rs.repo.Create(ctx, incoming)
+			if err == nil {
+				toUpdateOnStore = append(toUpdateOnStore, incoming)
 			}
-		} else {
-			incoming.MarkToCreate()
-			output = append(output, incoming)
+			me.Append(err)
+			continue
+		}
 
-			rs.logger.Info("resource [%s] will be created", incoming.FullName())
+		if resource.StatusIsSuccess(existing.Status()) && incoming.Equal(existing) {
+			_ = incoming.MarkSkipped()
+			rs.logger.Warn("resource [%s] is skipped because it has no changes", existing.FullName())
+			continue
 		}
+
+		if resource.StatusForToCreate(existing.Status()) {
+			_ = incoming.MarkToCreate()
+		} else if resource.StatusForToUpdate(existing.Status()) {
+			_ = incoming.MarkToUpdate()
+		}
+
+		err := rs.repo.Update(ctx, incoming)
```

Comment:
Maybe we can move this `Update` repo call into outside of this function `getResourcesToBatchUpdate`; or we change the name `getResourcesToBatchUpdate` into something else like `updateResourcesForDeploy`. Because if we don't do any changes, then the function will not reflect what it actually does.

Reply:
I disagree, let's discuss that.